cmd/agent/subcommands/remoteconfig/command.go

```diff
@@ -14,42 +14,53 @@ import (
 
 	"github.com/fatih/color"
 	"github.com/spf13/cobra"
-	"google.golang.org/grpc"
+	"go.uber.org/fx"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/protobuf/types/known/emptypb"
 
 	"github.com/DataDog/datadog-agent/cmd/agent/command"
-	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/config"
 	"github.com/DataDog/datadog-agent/pkg/api/security"
-	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 	agentgrpc "github.com/DataDog/datadog-agent/pkg/util/grpc"
 )
 
+// cliParams are the command-line arguments for this subcommand
+type cliParams struct {
+	*command.GlobalParams
+}
+
 // Commands returns a slice of subcommands for the 'agent' command.
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+	cliParams := &cliParams{
+		GlobalParams: globalParams,
+	}
 	remoteConfigCmd := &cobra.Command{
 		Use:   "remote-config",
 		Short: "Remote configuration state command",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			err := common.SetupConfigWithoutSecrets(globalParams.ConfFilePath, "")
-			if err != nil {
-				return fmt.Errorf("Unable to set up global agent configuration: %v", err)
-			}
-
-			if !config.Datadog.GetBool("remote_configuration.enabled") {
-				return fmt.Errorf("Remote configuration is not enabled")
-			}
-			return state(cmd, args)
+			return fxutil.OneShot(state,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfFilePath:      globalParams.ConfFilePath,
+					ConfigLoadSecrets: false,
+				}),
+				core.Bundle,
```

Comment:
Question: the function takes the `config` and then the `cliParams`. Does the order we supply arguments matters to `fx.New(opts...)`  ? I thought it did not.

If not could we invert the order to match the argument order of the function, I think it makes the code more intuitive to understand.

Reply:
The order does not matter, you're right.

I'm not sure what you mean by "the function", though?

Reply:
Oh, the one-shot function.  I'll just change the order of its arguments.